### PR TITLE
Fix crawling issue for some blogs 

### PR DESCRIPTION
**Pull request by IsAmrish**
Signed-off-by: isamrish <askmaurya48@gmail.com>

This PR will fix:
-- crawling issue for some blogs
-- use defaultImage if feature image ( of a particular blog) is not present

### CODE REVIEW

File: website/src/utils/isImageExist.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const isImageExist = (src: string) => {
```
Comment (Pallavi-PH) [inline website/src/utils/isImageExist.ts:1]:
```suggestion
export const doesImageExist = (src: string) => {
```

**Comment by rakeshPRaghu** [comment]
image
the default image is coming for the blogs, even though those having featured images
image
But the metadata is coming in as correctly

File: website/src/utils/doesImageExist.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const doesImageExist = (src: string) => {
+    const img = new Image();
```
Comment (AVRahul) [inline website/src/utils/doesImageExist.ts:2]: One line description for this util will be good to have.

File: website/src/pages/Blog/Author/index.tsx [hunk 3]
```diff
@@ -80,9 +88,21 @@ const Blog: React.FC = () => {
...
+      const url = `${currentOrigin}/blog/author/${authorName}`;
+      setAuthorMetadata({ author, author_info, image, url });
+    }
+  }, [filteredAuthorData, currentOrigin, authorName]);
+
   return (
     <>
-     <Metadata title={SeoJson.pages.blog.title} description={SeoJson.pages.blog.description} url={`${currentOrigin}${SeoJson.pages.blog.url}`} image={`${currentOrigin}${SeoJson.pages.blog.image}`} isPost={false} type="Series"  />
+     { authorMetadata && (
+       <Metadata title={authorMetadata?.author} description={authorMetadata?.author_info} url={authorMetadata?.url} image={authorMetadata?.image} isPost={false} type="Series"  />
```
Comment (AVRahul) [inline website/src/pages/Blog/Author/index.tsx:104]: can the type be derived from some constants file or something?

File: website/src/pages/BlogPage/index.tsx [hunk 4]
```diff
@@ -118,15 +119,15 @@ const BlogPage: React.FC = () => {
...
+  
   return (
     <>
     {currentBlogDetails?.id && (
       <Metadata 
         title={currentBlogDetails?.title} 
         description={currentBlogDetails?.excerpt} 
         url={currentLocation} 
-        image={`${currentOrigin}/images${currentPathname}.png` || `${currentOrigin}/images/blog/defaultImage.png`} 
+        image={`${currentOrigin}/images${currentPathname}.png`}
```
Comment (AVRahul) [inline website/src/pages/BlogPage/index.tsx:130]: We won't require the OR condition anymore??

File: website/src/hooks/useCurrentHost.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const useCurrentHost = () => {
+    const currentOrigin = window && window.location.origin;
```
Comment (AVRahul) [inline website/src/hooks/useCurrentHost.ts:2]: Some comments will be good to have here

File: website/src/pages/BlogPage/index.tsx (hunk 4 above)
Comment (IsAmrish) [reply website/src/pages/BlogPage/index.tsx:130]: yes

File: website/src/hooks/useCurrentHost.ts (hunk 5 above)
Comment (IsAmrish) [reply website/src/hooks/useCurrentHost.ts:2]: I believe, code is self explainatory!

File: website/src/utils/doesImageExist.ts (hunk 2 above)
Comment (IsAmrish) [reply website/src/utils/doesImageExist.ts]: removed it :)

**Comment by IsAmrish** [comment]
@rakeshPRaghu  

> the default image is coming for the blogs, even though those having featured images

This is fixed now :)

File: website/src/blogs/a-primer-on-openebs-continuous-integration.md [hunk 6]
```diff
@@ -32,7 +33,7 @@ The e2e phase involves running several parallel pipelines (based on Kubernetes c
 
 ![OpenEBS CI Workflow](/images/blog/openebs-ci-flow.jpg)
 
-**Note**: *Currently, the GitLab CI works in a “retrospective mode,” as it is invoked on commits to the upstream branches. There is work in progress to extend the support for pull requests (presently, a travis-based build verifies commit sanity to aid PR acceptance).*
+**Note**: _Currently, the GitLab CI works in a “retrospective mode,” as it is invoked on commits to the upstream branches. There is work in progress to extend the support for pull requests (presently, a travis-based build verifies commit sanity to aid PR acceptance)._
```
Comment (Pallavi-PH) [inline website/src/blogs/a-primer-on-openebs-continuous-integration.md:36]: Why this change? Using asterisks is the recommended method for text styling https://www.markdownguide.org/basic-syntax/

**Review by Pallavi-PH — APPROVED** [review]

**Review by rakeshPRaghu — APPROVED** [review]

**Review by AVRahul — APPROVED** [review]

File: website/src/pages/BlogPage/index.tsx [hunk 7]
```diff
@@ -118,15 +119,27 @@ const BlogPage: React.FC = () => {
...
+        description={currentBlogDetails?.excerpt} 
+        url={currentLocation} 
+        image={`${currentOrigin}/images/blog/defaultImage.png`} 
+        isPost={true}
+        author={{ name: currentBlogDetails?.author, image: `${currentOrigin}/images/blog/authors/${currentBlogDetails?.author
+        .toLowerCase().replace(/[^\w ]+/g,'')
+        .replace(/ +/g,'-')}.png` }}
+        tags={currentBlogDetails.tags}  />
+    )}
+    {currentBlogDetails?.id && !Boolean(currentBlogDetails?.hasFeatureImage) && (
```
Comment (rakeshPRaghu) [inline website/src/pages/BlogPage/index.tsx:137]: This is supposed to be `notHasFeatureImage` right?

Comment (IsAmrish) [reply website/src/pages/BlogPage/index.tsx]: yup, thanks for pointing out!!

**Review by satyapriyamishra222 — APPROVED** [review]
